<a href="https://colab.research.google.com/github/BilalAbuZahra/computer-vision/blob/master/lab01.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [5]:
x = 3
y = 10.5
Msg = 'Result'
fx = x+y * 10 ** 2
print ('The Result is ',fx)


The Result is  1053.0


mathimatical operation

In [6]:
print(f'The {Msg} is {fx}')

The Result is 1053.0


formatting the **output**

In [7]:
print (f'The modulo operation in Python is %, so for instance 10%3 is {10%3}')


The modulo operation in Python is %, so for instance 10%3 is 1


In [8]:
my_lst = [1,2,3,4,5,6,7,8,9,10]
print ('First element in the list is ',my_lst[0])
print ('Last element in the list is ',my_lst[len(my_lst)-1])

First element in the list is  1
Last element in the list is  10


In [9]:
sub_lst = my_lst[2:4]

sub_lst

[3, 4]

In [10]:
my_lst[:1]
my_lst[1:]
my_lst[-1:]
my_lst[:-1]

[1, 2, 3, 4, 5, 6, 7, 8, 9]

 first line is for calling all the elements from the first element for the begining of my list

---


 the second line for calling all the elements from the second element which means all numbers from 2 to the end of my list


---




In [11]:
msg = 'Welcome to Al Quds University'
print(msg[1])
print(f'first char of msg is {msg[0]}')

e
first char of msg is W


In [12]:
tuple_example = 1,2,4,10,24
print(tuple_example)

(1, 2, 4, 10, 24)


In [13]:
print(tuple_example[0])
print(tuple_example[:2])
print(tuple_example[:3])
print(tuple_example[1:])
print(tuple_example)

1
(1, 2)
(1, 2, 4)
(2, 4, 10, 24)
(1, 2, 4, 10, 24)


In [14]:
module_dict = {
"Code": "CS2021",
"Name": "Computer Vision",
"Year": 2020,
"Stage":3
}

In [15]:
print(module_dict)

{'Code': 'CS2021', 'Name': 'Computer Vision', 'Year': 2020, 'Stage': 3}


In [16]:
print(f'Module {module_dict["Code"]}')
print(f'Is {module_dict["Name"]}')
print(f'And will be taught to students of Stage {module_dict["Stage"]}')
print(f'And will be delivered in the year {module_dict["Year"]}')

Module CS2021
Is Computer Vision
And will be taught to students of Stage 3
And will be delivered in the year 2020


In [17]:
print(module_dict.get('Code'))

CS2021


In [18]:
module_dict["Year"]=2021
print(module_dict["Year"])

2021


In [19]:
for key, value in module_dict.items():
  print (key,':',value)

Code : CS2021
Name : Computer Vision
Year : 2021
Stage : 3


In [20]:
print(type(module_dict))
print(type(tuple_example))
print(type(my_lst))

<class 'dict'>
<class 'tuple'>
<class 'list'>


1.2 python Casting

In [21]:
float_v = 1.1
int_v = 1
string_v = 'this is string'

In [22]:
print(type(float_v))
print(type(int_v))
print(type(string_v))

<class 'float'>
<class 'int'>
<class 'str'>


In [23]:
int_v = float(int_v)
print('The value of int_v is ', int_v)
print('The Type of int_v after casting is ',type(float_v))

The value of int_v is  1.0
The Type of int_v after casting is  <class 'float'>


1.3 Logical operations


In [24]:
print(1 ==1)
print(10 < 9)
print(10 > 9)
print(10 >= 9)

True
False
True
True


In [25]:
small = 10
big = 100
results = ((small <big) and (big%small==0))
results1 = ((small <big) and (big%small!=0))
results2 = ((small <big) or (big%small!=0))
results3 = ((small >big) or (big%small!=0))

In [26]:
print(f'({small} <{big}) and ({big}%{small}==0) ', results)
print(f'({small} <{big}) and ({big}%{small}!=0) ',results1)
print(f'({small} <{big}) or ({big}%{small}!=0) ',results2)
print(f'({small} >{big}) or ({big}%{small}!=0) ',results3)

(10 <100) and (100%10==0)  True
(10 <100) and (100%10!=0)  False
(10 <100) or (100%10!=0)  True
(10 >100) or (100%10!=0)  False


1.4 conditions and Loops

In [27]:
a = 33
b = 200
if b > a:
  print('b is greater than a because ', f"{b} is greater than {a}")
else:
  print(f"{b} is greater than {a}")


b is greater than a because  200 is greater than 33


In [28]:
a = 33
b = 200
if b > a:
 print('b is greater than a because ', f"{b} is greater than {a}")
else:
 print(f"{b} is greater than {a}")

b is greater than a because  200 is greater than 33


In [29]:
if a > b: print("a is greater than b")

In [30]:
a = 200
b = 33
c = 500
# both conditions must be true
if a > b and c > a:
 print("Both conditions are True")

Both conditions are True


In [31]:
a = 200
b = 33
c = 500
if a > b or a > c:
 print("At least one of the conditions is True")

At least one of the conditions is True


In [32]:
i = 1
while i < 10:
 print(f'{i} to the power 2 is {i**2}')
 i += 1

1 to the power 2 is 1
2 to the power 2 is 4
3 to the power 2 is 9
4 to the power 2 is 16
5 to the power 2 is 25
6 to the power 2 is 36
7 to the power 2 is 49
8 to the power 2 is 64
9 to the power 2 is 81


In [33]:
i = 1
while i < 10:
 print(f'{i} to the power 2 is {i**2}')
 i += 1
 if i>5:
  break


1 to the power 2 is 1
2 to the power 2 is 4
3 to the power 2 is 9
4 to the power 2 is 16
5 to the power 2 is 25


In [34]:
i = 0
while i < 10:
  i += 1 # i = i + 1
  if i==5:# skip five
   continue
  print(f'{i} to the power 2 is {i**2}')


1 to the power 2 is 1
2 to the power 2 is 4
3 to the power 2 is 9
4 to the power 2 is 16
6 to the power 2 is 36
7 to the power 2 is 49
8 to the power 2 is 64
9 to the power 2 is 81
10 to the power 2 is 100


In [35]:
ds_tools = ["Python", "R", "Git"]
for tool in ds_tools:
 print('Data Science Tool: ', tool)


Data Science Tool:  Python
Data Science Tool:  R
Data Science Tool:  Git


In [36]:
for i in range(10):
 print(f'square of {i} is {i**2}')

square of 0 is 0
square of 1 is 1
square of 2 is 4
square of 3 is 9
square of 4 is 16
square of 5 is 25
square of 6 is 36
square of 7 is 49
square of 8 is 64
square of 9 is 81


In [37]:
for i in 'The University':
 print(i)
 if i == ' ':
  break

T
h
e
 


1.5 Functions

In [38]:
def power_func(x):
 results = x **2
 results = results * 2 + 5
 return results

In [39]:
power_func(5)

55

In [40]:
def cubic_x(x,y):
 return x ** y + 5

In [41]:
x = 3
y = 2
print(f'{x}^{y} + 5 is {cubic_x(x,y)}')

3^2 + 5 is 14


In [42]:
def my_course(*courses):
 print("I am considering the following courses " + courses[1])
my_course("CS", "DS", "Eng")

I am considering the following courses DS


1.6 Python Lambda

In [43]:
x = lambda a : a + 10
print(x(5))


15


In [44]:
x = lambda a, b : a * b
print(x(5, 6))

30


1.7  Import Statement

In [45]:
import pandas as pd
import numpy as np

In [46]:
import math
pi = math.pi
print(pi)

3.141592653589793


1.8  Introduction to Pandas and Numpy


In [47]:
import pandas as pd
import numpy as np

In [48]:
df = pd.read_csv('/content/Auto.csv')


**load** **file**

In [49]:
df.head()


mpg  cylinders  displacement horsepower  weight  acceleration  year  \
0  18.0          8         307.0        130    3504          12.0    70   
1  15.0          8         350.0        165    3693          11.5    70   
2  18.0          8         318.0        150    3436          11.0    70   
3  16.0          8         304.0        150    3433          12.0    70   
4  17.0          8         302.0        140    3449          10.5    70   

   origin                       name  
0       1  chevrolet chevelle malibu  
1       1          buick skylark 320  
2       1         plymouth satellite  
3       1              amc rebel sst  
4       1                ford torino

In [50]:
print(f'The size of df is {df.shape}')

The size of df is (397, 9)


the size of this data frame is 397 rows and 9 columns

In [51]:
df.columns

Index(['mpg', 'cylinders', 'displacement', 'horsepower', 'weight',
       'acceleration', 'year', 'origin', 'name'],
      dtype='object')

In [52]:
type(df['mpg'])

pandas.core.series.Series

In [53]:
cars70 = df[df.year==70]
cars70.head()


mpg  cylinders  displacement horsepower  weight  acceleration  year  \
0  18.0          8         307.0        130    3504          12.0    70   
1  15.0          8         350.0        165    3693          11.5    70   
2  18.0          8         318.0        150    3436          11.0    70   
3  16.0          8         304.0        150    3433          12.0    70   
4  17.0          8         302.0        140    3449          10.5    70   

   origin                       name  
0       1  chevrolet chevelle malibu  
1       1          buick skylark 320  
2       1         plymouth satellite  
3       1              amc rebel sst  
4       1                ford torino

In [54]:
print(f'Number of cars in cars70 DataFrame is {cars70.shape[0]}')

Number of cars in cars70 DataFrame is 29
